include/flatbuffers/idl.h

```diff
@@ -26,6 +26,14 @@
 #include "flatbuffers/hash.h"
 #include "flatbuffers/reflection.h"
 
+#if __GNUC__ >= 4
```

Comment:
Maybe put the changes in this file in its own header? they have no dependencies on the rest of idl.h, and if it is its own header, then it can be included/parsed as a C header.